memphis/memphis.py

```diff
@@ -255,19 +280,24 @@ def __init__(self, connection, producer_name, station_name):
         self.producer_name = producer_name
         self.station_name = station_name
 
-    async def produce(self, message, ack_wait_sec=15):
+    async def produce(self, message, ack_wait_sec=15, headers={}):
         """Produces a message into a station.
         Args:
             message (Uint8Array): message to send into the station.
             ack_wait_sec (int, optional): max time in seconds to wait for an ack from memphis. Defaults to 15.
+            headers ({}, optional): Message headers, defaults to {}.
         Raises:
             Exception: _description_
             Exception: _description_
         """
         try:
+            memphis_headers={ 
+            "$memphis_producedBy": self.producer_name, 
+            "$memphis_connectionId": self.connection.connection_id}
+            memphis_headers.update(headers)
```

Comment:
update the headers and not the memphis headers since at the end you are sending the headers and not only the memphis headers. this is more semantic than logic